components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -12853,53 +12857,105 @@ public void addAPIProduct(APIProduct apiproduct) throws APIManagementException {
             prepStmtAddAPIProduct.setString(7, apiproduct.getSubscriptionAvailability());
             prepStmtAddAPIProduct.setString(8, apiproduct.getUuid());
             prepStmtAddAPIProduct.execute();
-            connection.commit(); //TODO check this///////
-            
-            //Get the product id related to the product
-            //TODO check this
-            //TODO move to constant
-            String queryGetAPIProduct = "SELECT API_PRODUCT_ID FROM AM_API_PRODUCT WHERE UUID = ?";
-            
-            prepStmtGetAPIProduct = connection.prepareStatement(queryGetAPIProduct);
-            prepStmtGetAPIProduct.setString(1, apiproduct.getUuid());
-            rs = prepStmtGetAPIProduct.executeQuery();
+
+            rs = prepStmtAddAPIProduct.getGeneratedKeys();
+
             if (rs.next()) {
-                productId = rs.getInt("API_PRODUCT_ID");
+                productId = rs.getInt(1);
             }
             //breaks the flow if product is not added to the db correctly
             if(productId == 0) {
-                throw new APIManagementException("Error while retrieving API product " + apiproduct.getUuid()
-                        + " after inserting it to the database");
+                throw new APIManagementException("Error while adding API product " + apiproduct.getUuid());
             }
+
+            //add product scope
+            //TODO finalize format and move to constants
+            String productScopeKey = "productscope-" + apiproduct.getName() + ":" +apiproduct.getProvider();
+            //for now use key for display name as well TODO check and modify
+            String productScopeDisplayName = productScopeKey;
+            Scope productScope = new Scope();
+            productScope.setKey(productScopeKey);
+            productScope.setName(productScopeDisplayName);
+
+            String queryAddScopeEntry = SQLConstants.ADD_SCOPE_ENTRY_SQL;
+            prepStmtAddScopeEntry = connection.prepareStatement(queryAddScopeEntry, new String[]{"scope_id"});
+            prepStmtAddScopeEntry.setString(1, productScope.getKey());
+            prepStmtAddScopeEntry.setString(2, productScope.getName());
+            prepStmtAddScopeEntry.setString(3, productScope.getDescription());
+            prepStmtAddScopeEntry.setInt(4, tenantID);
+            prepStmtAddScopeEntry.execute();
+
+            rs = prepStmtAddScopeEntry.getGeneratedKeys();
+
+            if (rs.next()){
+                scopeId = rs.getInt(1);
```

Comment:
shall we use the column instead of the column id.

Reply:
Here I am retrieving the autogenerated scope id key. Autogenerated key column value has to be accessed by column index.